lib/Event/Model/Asset/ResolveUploadTargetEvent.php

```diff
@@ -38,6 +38,12 @@ public function __construct(int $parentId, string $filename, array $context)
     {
         $this->parentId = $parentId;
         $this->filename = $filename;
+        trigger_deprecation(
+            'pimcore/pimcore',
+            '11.5.0',
+            'The context property is deprecated and will be removed in 12.0.0.
+            Use setArgument() from the ArgumentsAwareTrait instead.'
+        );
```

Comment:
This deprecation is triggered every time this event object is created. I don't think this is how it should be done. Shouldn't it only be triggered if the `$context` parameter was actually passed?

Reply:
Since `$context` is not optional. It has to be passt every time as an argument. When we remove the `$context` in the future this will affect the creation of the Object. That's why we trigger the deprecation of here. 

Do you have any better suggestions?

Reply:
Yes: set it via `setArgument()` as you suggest in your deprecation message.

Reply:
There should be a possibility to actually fix the deprecation, so that the users are deprecation-free before updating to Pimcore 12.

Reply:
> There should be a possibility to actually fix the deprecation, so that the users are deprecation-free before upgrading to Pimcore 12.

Now i get your Point! Makes total sense. I will then make `$context` nullable and add the check.